benchmark/scripts/run.ts

```diff
@@ -0,0 +1,170 @@
+import {
+  CloudWatchLogsClient,
+  DeleteLogGroupCommand,
+  DescribeLogStreamsCommand,
+  GetLogEventsCommand,
+  GetLogEventsCommandInput,
+  OutputLogEvent,
+} from "@aws-sdk/client-cloudwatch-logs";
+import { InvokeCommand, LambdaClient } from "@aws-sdk/client-lambda";
+import { execSync } from "child_process";
+import { promises as fsPromises } from "fs";
+import path from "path";
+import { CONFIG } from "../benchmarkConfig";
+
+interface IResult {
+  git: {
+    branch: string;
+    tag?: string;
+  };
+  time: { start: string; end: string };
+  log: Record<string, any>[];
+}
+
+function wait(ms: number) {
+  return new Promise<void>((resolve) => {
+    setTimeout(() => {
+      resolve();
+    }, ms);
+  });
+}
+
+async function main() {
+  // Setup
+  const gitBranch = execSync("git rev-parse --abbrev-ref HEAD 2>/dev/null", {
+    encoding: "utf-8",
+  }).trim();
+  let gitTag: string | undefined;
+  try {
+    gitTag = execSync("git describe --tags --abbrev=0 2>/dev/null", {
+      encoding: "utf-8",
+    }).trim();
+  } catch {
+    gitTag = undefined;
+  }
+  const filePath = path.join(
+    __dirname,
+    CONFIG.logs.outputFolder,
+    `${gitBranch}/raw.json`
+  );
+  const rawResult: IResult = {
+    git: {
+      branch: gitBranch,
+      tag: gitTag,
+    },
+    time: { start: new Date().toISOString(), end: "" },
+    log: [],
+  };
+
+  console.log(
+    `Generate logs for branch '${gitBranch}' ${gitTag ? `(${gitTag})` : ""}`
+  );
+
+  console.log("\n- Delete previous artifacts if they exist");
+  // Delete if exist
+  // - previous results files
+  try {
+    console.log(`  - Previous result file: '${filePath}'`);
+    await fsPromises.unlink(filePath);
+  } catch {
+    console.log(`    - not found`);
+  }
+  // - previous cloudwatch log groups
+  for (const fn of CONFIG.functions) {
+    for (let i = 1; i <= CONFIG.runs; i++) {
+      const logGroupName = `/aws/lambda/${fn.functionName}-${i}`;
+      const command = new DeleteLogGroupCommand({ logGroupName });
+      console.log(`  - log group '${logGroupName}'`);
```

Comment:
why are you deleting cloudwatch logs?

Reply:
I flush any logs from potential previous runs
will be useful to prevent redeploying the benchmark stack across features/branche if it has not changed